clang/lib/CConv/ConstraintVariables.cpp

```diff
@@ -447,10 +456,12 @@ void PointerVariableConstraint::insertQualType(uint32_t TypeIdx,
     QualMap[TypeIdx].insert(RestrictQualification);
 }
 
-bool PointerVariableConstraint::emitArraySize(std::ostringstream &Pss,
+
+bool PointerVariableConstraint::emitArraySize(std::stack<std::string> &CheckedArrs,
                                               uint32_t TypeIdx,
                                               bool &EmitName,
-                                              bool &EmittedCheckedAnnotation,
+                                              bool &AllArrays,
+                                              bool &ArrayRun,
```

Comment:
Can you explain to me what these arguments are? Maybe add a comment to the top of the function, explaining them. It's a bit odd to have a big pile of booleans. Are all combinations of those booleans possible, or does (say) true for one imply true for another?

Reply:
Comment added to where the flags are declared.